template/cordova/lib/prepare.js

```diff
@@ -439,6 +490,109 @@ function getUAPVersions(config) {
+ * @param targetName 'SplashScreen'|'SplashScreenPhone'
+ * @returns boolean */ 
+function splashScreenTargetIs(splash, targetName) {
+    var matchesTarget, targetImg;
+
+    if (splash.target) {
+        // MRT syntax:
+        matchesTarget = splash.target === targetName;
+    } else {
+        // Fall back on find by size for old non-MRT syntax:
+        targetImg = findPlatformImage(splash.width, splash.height);
+        matchesTarget = splashScreenTargetProjectMatchesTargetName(targetImg.targetProject, targetName);
+    }
+
+    return matchesTarget;
+}
+
+// Updates manifests to match the app splash screen image types (PNG/JPG/JPEG)
+function updateSplashScreenImageExtensions(cordovaProject, locations) {
+
+    // Saving all extensions used for targets to verify them later
+    var extensionsUsed = {};
+
+    function checkThatExtensionsAreNotMixed() {
+        for (var target in extensionsUsed) {
+            /*jshint loopfunc: true */
+            if (extensionsUsed.hasOwnProperty(target)) {
+                var extensionsUsedForTarget = extensionsUsed[target];
+
+                // Check that extensions are not mixed:
+                if (extensionsUsedForTarget.length > 1 && extensionsUsedForTarget.some(function(item) {
+                    return item !== extensionsUsedForTarget[0];
+                })) {
+                    events.emit('warn', '"' + target + '" splash screens have mixed file extensions which is not supported. Some of the images will not be used.');
+                }
+            }
+        }
+    }
+
+    function checkTargetMatchAndUpdateUsedExtensions(img, target) {
+        var matchesTarget = splashScreenTargetIs(img, target);
+
+        if (matchesTarget === true) {
+            extensionsUsed[target] = extensionsUsed[target] || [];
+            extensionsUsed[target].push(path.extname(img.src.toLowerCase()));
+        }
+
+        return matchesTarget;
+    }
+
+    function updateSplashExtensionInManifest(manifestFileName, splashScreen) {
+        var manifest = AppxManifest.get(path.join(locations.root, manifestFileName));
+        var newExtension = path.extname(splashScreen.src);
+
+        if (manifest.getVisualElements().getSplashScreenExtension() !== newExtension) {
+            events.emit('verbose', 'Set ' + manifestFileName + ' SplashScreen image extension to "' + newExtension + '"');
+            manifest.getVisualElements().setSplashScreenExtension(newExtension);
+            manifest.write();
+        }
+    }
+
+    var splashScreens = cordovaProject.projectConfig.getSplashScreens('windows');
+
+    var desktopSplashScreen = splashScreens.filter(function(img) { 
+        return checkTargetMatchAndUpdateUsedExtensions(img, SPLASH_SCREEN_DESKTOP_TARGET_NAME);
+    })[0];
+
+    var phoneSplashScreen = splashScreens.filter(function(img) { 
+        return checkTargetMatchAndUpdateUsedExtensions(img, SPLASH_SCREEN_PHONE_TARGET_NAME);
+    })[0];
+
+    checkThatExtensionsAreNotMixed();
```

Comment:
> You should not mix image extensions for non-MRT syntax inside one project group

I thought it's pretty obvious :)

> in this case ^ the second JPG image will not be used as AppxManifest will have PNG for SplashScreen.Image' extension

Yep. If you specify the same splascreens using MRT and 'target' attribute, `splashscreen.scale-125.jpg` also would be silently ignored because `mapImageResources` requires extensions to match. That said I don't think we need to care about this at all.